### PR TITLE
CTM360-hv-cbs-integration

### PR DESCRIPTION
**Pull request by edx-sayed-salman**
Required items, please complete
   
   Change(s):
   - Added new solution and support CCP connector.

   Reason for Change(s):
   - Added new solution and support CCP connector.

   Version Updated:
   - NA

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - NA

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Thanks for raising this PR. This PR will be investigated and will update you about the same before 30 October, 2023.
Until then you can repackage the solution using V3 tool. 
Instructions on how to use V3 tool: https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman Please repackage the solution using V3 tool and any reason behind removing the Data file.

**Comment by edx-sayed-salman** [comment]
Hello @v-prasadboke , according to the [document](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions#step-2--package-your-content) the the 'Data' file creation it's in the step 2, not in step 1 , so i decided to remove it, also one check failed which is related to unresolved link inside the data file which is the logo.

after repackage the solution, should i push it with this request or only through [Microsoft Partner Center](https://docs.microsoft.com/partner-center/overview).?

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Please add the Data file again. Its needed to repackage the solution. Please add it again I'll take a look at it.

**Comment by v-prasadboke** [comment]
To push the package to partner First it needs approval by the reviewer from Github PR and should be in master branch.

**Comment by edx-sayed-salman** [comment]
@v-prasadboke please can you have alook.

**Comment by v-prasadboke** [comment]
Hey Sayed, Thanks for providing the requested changes. Will take a look at it and will update you before 03 November, 2023.
Also can you please provide me write access to your branch.

**Comment by edx-sayed-salman** [comment]
@v-prasadboke  we give you write access

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, I've added the missing fields which are required for the detections. please add the required data connector and data type fields which are empty. 
Thanks.

**Comment by edx-sayed-salman** [comment]
Hi @v-prasadboke can you have alook at updated files

**Comment by v-prasadboke** [comment]
Hello Sayid Salman thank you for committing the required changes. Will take a look at it.

File: Solutions/CTM360/Analytic Rules/HeaderXContentTypeOptionsMissing.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+id: e6fe772a-c669-4b42-b25a-e01ab8ab02f7
+name: 'Header: X-Content-Type-Options Missing'
+severity: Medium
```
Comment (v-prasadboke) [inline Solutions/CTM360/Analytic Rules/HeaderXContentTypeOptionsMissing.yaml:3]: Hello @edx-sayed-salman description is missing from this detection due to which its failing for solution packaging

Comment (edx-sayed-salman) [reply Solutions/CTM360/Analytic Rules/HeaderXContentTypeOptionsMissing.yaml:3]: Hi , @v-prasadboke , we add the description in the rule.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman. Thanks for committing the required changes. I'll take a look at it and respond back with some updates. 
Thank you.

**Comment by edx-sayed-salman** [comment]
Hello, @v-prasadboke I have some questions I'd like to ask.

I encountered an error after deploying to Azure Sentinel, which appears to be related to the rule template. The rule is displayed as follows:
image

The specific error message I received is: "Failed to save analytics rule 'Phishing'. BadRequest: Invalid data model. [Properties.IncidentConfiguration.GroupingConfiguration.LookbackDuration: The field LookbackDuration must be between PT5M and P7D.]"

I checked the time settings, and they appear to be within the specified range. I also searched the internet for a solution but couldn't find one.

Additionally, from what I've read, the rule should be enabled after deployment. I have approximately 40 rule templates. Is there a way to make enabling this rule a one-click process for the user?

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Sorry for the inconvenience but due to lack of availability I was unable to test the PR. 
I'll review this PR and will take a look at your issue. Thanks.

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, Currently V3 tool doesn't supports CCP data connector. Packaging tool team is working on it. 
We will update you once its done. Until then can you please resolve validation errors,

**Comment by edx-sayed-salman** [comment]
Hi @v-prasadboke , now the validation should be fix and i have used the V2 to pack the connector can you finalize this ?, thank you

**Comment by v-prasadboke** [comment]
Sorry @edx-sayed-salman, but we have stopped using V2 for packaging. Our Packaging tool team is working on the tool to get this CCP connector in the maintemplate.

**Comment by edx-sayed-salman** [comment]
Dear @v-prasadboke , thanks for your reposne , do you know when this will be ready , since now i need to do it as soon as possible, also i am working on azure function to do the pull and send data to sentinel is V3 will work for azure function?

**Comment by v-prasadboke** [comment]
Hi Sayed Repackaging team has worked on the CCP connector. I will share the requirements with you in a bit. 
If possible can we connect now, Need some understanding and also I will elaborate the requirements for ccp connector.

**Comment by edx-sayed-salman** [comment]
Hi @v-prasadboke sure , send me the requirement. how do we connect now dear ? is there anyway for direct communication

**Comment by v-prasadboke** [comment]
Hello @edx-sayed-salman, do we have any updates for ccp connector

**Comment by edx-sayed-salman** [comment]
@v-prasadboke i dont have update yet , i have other PR as azure function app , if that finish will start the CCP connector also , thank you

**Comment by v-prasadboke** [comment]
Until then can you put this PR in draft state

**Comment by v-prasadboke** [comment]
We wanted to check on the status of PR #9276 . PR is pending for more than x days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 5 business days, we will close this PR. Thank you for your cooperation.

**Pull request closed without merging**